Metadata quality verification API (#1874)
quiltdata/quilt: docs/Advanced Features/workflows.md

> *New in Quilt 3.3*

Author: @akarve 
could you check this?
Config and schemas are in the bucket, so all these commands work and you can play with it if you want.

> `quilt3` provides API to ensure quality of package metadata. To start use it you need to create

Reviewer: ```suggestion
`quilt3` provides an API to ensure the quality of package metadata. To get started, create a
```

> configuration file in your registry under `.quilt/workflows/config.yml`. For example:

Reviewer: Shouldn't we provide this as a constant from `quilt3`, or from an API?

Author: I think that currently there is no good place for it.

Reviewer: What about util.py? See https://docs.quiltdata.com/api-reference/cli#constants-see-util-py-for-more

```yaml
version: "1"
workflows:
  alpha:
    name: Search for aliens
    is_message_required: true
  beta:
    name: Studying superpowers
    metadata_schema: superheroes
  gamma:
    name: Nothing special
    description: TOP SECRET
    is_message_required: true
    metadata_schema: top-secret
schemas:
 superheroes:
   url: s3://quilt-sergey-dev-metadata/schemas/superheroes.schema.json
 top-secret:
   url: s3://quilt-sergey-dev-metadata/schemas/top-secret.schema.json
```

> With this configuration you can't push anything without specifying workflow:

Reviewer: ```suggestion
With the above configuration you must specify a workflow before you can push:
```

```python
>>> import quilt3
>>> quilt3.Package().push('test/package', registry='s3://quilt-sergey-dev-metadata')

> QuiltException: Workflow is required, but none specified.

Reviewer: ```suggestion
QuiltException: Workflow required, but none specified.
```
